core/nixpkgs.bzl

```diff
@@ -124,9 +125,25 @@ Create an external repository representing the content of Nixpkgs, based on a Ni
 """,
 )
 
+def _nixopts_args(nixopts):
+    result = {}
+    arg_opt, arg_name = None, None
+    for opt in nixopts:
+        if opt == "--arg" or opt == "--argstr":
+            arg_opt, arg_name = opt, None
+        elif arg_opt:
+            if arg_name == None:
+                arg_name = opt
+            else:
+                arg_val = opt if arg_opt == "--arg" else "''%s''" % opt
```

Comment:
Could this trigger quoting issues if a users passes in a quoted `--argstr` option?

Reply:
Possibly, and now that I think about it, it may end up removing spaces in the user string. Should probably use `"` instead, and manually escape `"`, `\` and `${`